### PR TITLE
UI tests for moving files & folders

### PR DESCRIPTION
**Pull request by individual-it**
## Description
UI tests that check if moving of files and folders happens correctly in the Web UI

## Related Issue
owncloud/QA#437

## Motivation and Context
run smoke tests automatically

## How Has This Been Tested?
run new tests locally, travis will tests the whole set

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: tests/ui/features/bootstrap/FilesContext.php [hunk 1]
```diff
@@ -326,6 +399,58 @@ public function theFileFolderShouldNotBeListed($name, $pageObject = null) {
...
+		$this->theFileFolderShouldBeListed($itemToBeListed);
+	}
+
+	/**
+	 * @Then /^the moved elements should (not|)\s?be listed in the in the folder ['"](.*)['"]$/
+	 * @param string $shouldOrNot
+	 * @param string $folderName
+	 * @return void
+	 */
+	public function theMovedElementsShouldBeListedInTheInTheFolder(
```
Comment (phil-davis) [inline tests/ui/features/bootstrap/FilesContext.php]: InTheInThe
in the step text as well as the method name.
search/replace

File: tests/ui/features/files/moveFiles.feature [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		When I batch move these files into the folder "simple-empty-folder"
+		| name        |
+		| data.zip    |
+		| lorem.txt   |
+		| testapp.zip |
+		Then the moved elements should not be listed
+		And the moved elements should not be listed after a page reload
+		But the moved elements should be listed in the in the folder "simple-empty-folder"
+
+	Scenario: move a file into a folder (problematic characters)
```
Comment (phil-davis) [inline tests/ui/features/files/moveFiles.feature]: This is a fun test :)

File: tests/ui/features/files/moveFolders.feature [hunk 3]
```diff
@@ -0,0 +1,57 @@
+Feature: moveFolders
+
+	Background:
+		Given a regular user exists
+		And I am logged in as a regular user
+		And I am on the files page
+
+	Scenario: move a folder into an other folder
```
Comment (phil-davis) [inline tests/ui/features/files/moveFolders.feature]: Since you have to make some changes, may as well:
s/an other/another/

**Review by phil-davis — CHANGES_REQUESTED** [review]

**Review by phil-davis — APPROVED** [review]

**Comment by individual-it** [comment]
backport to stable10 in #29062

**Pull request merged**